File: elixir/lib/player_strategy.ex
defmodule PlayerStrategy do
  @callback choose_free_feed(
              game_state :: GameState.t(),
              player_name :: binary(),
              feeds :: list()
            ) :: Feed.t()

  @callback choose_put_back_feed(
              game_state :: GameState.t(),
              player_name :: binary(),
              feeds :: list()
            ) :: Feed.t()

  @callback choose_buy_feed(
              game_state :: GameState.t(),
              player_name :: binary(),
              gem_count :: integer()
            ) :: Feed.t()
end

File: kotlin/src/main/kotlin/PlayerStrategy.kt
package com.sheepduke.donkeyrace

interface PlayerStrategy {
    fun chooseFreeFeed(player: Player, feeds: List<Feed>, state: GameState, context: GameContext): Feed

    fun chooseFeedToPutBack(player: Player, state: GameState, context: GameContext): Feed

    fun buyFeeds(player: Player, state: GameState, context: GameContext): List<Feed>
}